Common/Securities/Security.cs

```diff
@@ -625,6 +639,10 @@ public void SetRealTimePrice(BaseData data)
         /// <param name="leverage">Leverage for this asset</param>
         public void SetLeverage(decimal leverage)
         {
+            if (Symbol.ID.SecurityType == SecurityType.Future ||
+                Symbol.ID.SecurityType == SecurityType.Option)
+                return;
```

Comment:
In this case we throw/catch exception every time system calls SetLeverage in security class (which happens e.g. in initializers). Expensive operation. The idea of setting leverage is specific to asset classes (FX, equities) and retail brokers' business model (generally you can't set the leverage in most cases), so it makes sense to add  CanSetLeverage() method.